drafts/npm.md

```diff
@@ -0,0 +1,472 @@
+   shrinkwrap.json should ***not*** be published. The reasoning is that version
+   resolution should be left to the package consumer. Allow all versions from
+   the minimum to the latest you support, e.g., `^m.n.o` to pin to a major
+   range; `~m.n.o` to pin to a minor range. Avoid versions with critical
+   vulnerabilities as much as possible. Visit the [semver
+   calculator](https://semver.npmjs.com/) to help you define the ranges.
+
+### Maintenance
+
+It is important to update dependencies periodically, in particular when new
+vulnerabilities are disclosed and patched. Tools that help with dependency
+management are easy to use and may implement security checks for you.
+
+**Recommendations:**
+
+1. In order to manage your dependencies, use a tool such as
+   [dependabot](https://docs.github.com/en/code-security/supply-chain-security/managing-vulnerabilities-in-your-projects-dependencies/configuring-dependabot-security-updates)
+   or [renovatebot](https://github.com/renovatebot/renovate). These tools
+   submit merge requests that you may review and merge into the default branch.
+
+1. Stay up to date about existing vulnerabilities in your dependencies:
+
+    1. If you have installed the tools above, enable security alerts: see
+       [dependabot
+       config](https://docs.github.com/en/code-security/supply-chain-security/managing-vulnerabilities-in-your-projects-dependencies/about-alerts-for-vulnerable-dependencies#dependabot-alerts-for-vulnerable-dependencies)
+       and [renovatebot
+       config](https://docs.renovatebot.com/configuration-options/#vulnerabilityalerts). Note
+       that renovatebot follows config-as-code, so it makes it easier for
+       external users to verify it is enabled.
+
+    1. If you prefer a dedicated tool, run
+       [npm-audit](https://docs.npmjs.com/cli/v8/commands/npm-audit)
+       periodically, e.g., in a GitHub workflow.
+
+1. In order to remove dependencies, periodically run [`npm
+   prune`](https://docs.npmjs.com/cli/v8/commands/npm-prune) and submit a merge
+   request. Tools above do not support this feature yet, and we are not aware
+   of a GitHub action for this feature.
+
+## Release
+
+### Account
+
+Publishing on the npm registry requires creating a user account.
+
+**Recommendations:**
+
+1. Developers should [enable
+  2FA](https://docs.npmjs.com/configuring-two-factor-authentication) on their
+  account.
+
+### Signing and Verification
+
+npm packages are all signed by a [PGP
+key](https://docs.npmjs.com/about-pgp-signatures-for-packages-in-the-public-registry)
+owned by the default npm registry.
+
+**Warning**: npm does not support user-level signatures.
+
+### Publishing
+
+Publishing a package uploads it to a registry for others to install and
+download.
+
+**Recommendations:**
+
+1. Use an [Automation
+   token](https://docs.npmjs.com/creating-and-viewing-access-tokens) to
+   authenticate and publish to the default npm registry.
```

Comment:
this is strictly less secure than a human doing it locally, because it's a single factor, instead of the two-factor that human publishes require.

Reply:
Good point. Let's split this into different sections for "manual" and "automated".